### PR TITLE
feat: add long-audio retranscription and decoder hardening

### PR DESCRIPTION
**Pull request by ysdede**
Summary
- add sentence-aware transcribeLongAudio() retranscription with chunk output support
- harden decoder validation, cache hashing, and tensor cleanup paths
- document the new API and add regression coverage for long-audio, timestamps, confidences, and error paths

Validation
- npm test

### CODE REVIEW

File: src/long_audio.js [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+    timeOffset = 0,
+    ...transcribeOptions
+  } = opts;
+
+  const wantWordTimestamps = returnTimestamps === 'word';
+  const wantTimestampChunks = returnTimestamps === true || wantWordTimestamps;
+  const normalizedChunkLengthS = normalizeChunkLengthS(chunkLengthS);
+  const audioDurationS = audio.length / sampleRate;
+  const autoWindowing = normalizedChunkLengthS <= 0 && audioDurationS > AUTO_WINDOW_THRESHOLD_S;
+  const effectiveChunkLengthS = normalizedChunkLengthS > 0
```
[the remark this answers is not in the document]
Comment (ysdede) [reply src/long_audio.js]: Applied. 	ranscribeLongAudioWithChunks() now validates sampleRate before any duration math and rejects invalid 	imeOffset before scheduling windows. I also added a regression test to confirm we fail early without calling model.transcribe().

File: src/parakeet.js [hunk 2]
```diff
@@ -285,19 +297,47 @@ export class ParakeetModel {
...
 
     // subarray(): zero-copy view into joiner output buffer.
     // Do NOT mutate tokenLogits/durLogits without copying first (.slice()).
     const tokenLogits = data.subarray(0, vocab);
     const durLogits = data.subarray(vocab, totalDim);
+    if (durLogits.length === 0) {
+      logits.dispose?.();
+      this._disposeDecoderState({ state1: outputState1, state2: outputState2 }, currentState);
+      throw new Error('ParakeetModel decoder output is missing required TDT duration logits.');
+    }
```
[the remark this answers is not in the document]
Comment (ysdede) [reply src/parakeet.js]: Applied. _runCombinedStep() now disposes the just-returned decoder tensors and the active decoder state before throwing on malformed outputs, and there is a regression test covering the missing-duration-logits path to make sure those tensors are released.
